src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -115,20 +117,96 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
             var originalTemplatePath = tokenFromOriginalTemplate.Path;
             var unmatchedPathInOriginalTemplate = pathInExpandedTemplate[originalTemplatePath.Length..].TrimStart('.');
 
+            bool pathsAreEqual = string.Equals(originalTemplatePath, pathInExpandedTemplate, StringComparison.OrdinalIgnoreCase);
+
+            bool unmatchedSegmentReferencesResourcesArray = unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase);
+            bool unmatchedSegmentStartsWithIndexArray = unmatchedPathInOriginalTemplate.StartsWith("[", StringComparison.OrdinalIgnoreCase);
+            bool originalPathEndsWithResourcesArray = originalTemplatePath.EndsWith("resources", StringComparison.OrdinalIgnoreCase);
+
+            bool pathIsNotFromCopiedResource = !unmatchedSegmentReferencesResourcesArray
+                    && !unmatchedSegmentStartsWithIndexArray
+                    && !originalPathEndsWithResourcesArray;
+
             // Compare the path of the expanded template with the path of the JToken found in the original template.
-            // If they match, or if the first unmatched property in the original is NOT a resources array,
+            // If they match or if the path is not from a copied resource
             // the line number from the JToken of the original template can be returned directly.
-            if (originalTemplatePath.Length == pathInExpandedTemplate.Length
-                || !unmatchedPathInOriginalTemplate.StartsWith("resources", StringComparison.OrdinalIgnoreCase))
+            if (pathsAreEqual || pathIsNotFromCopiedResource)
             {
                 return (tokenFromOriginalTemplate as IJsonLineInfo)?.LineNumber ?? 0;
             }
 
-            // The first unmatched property must be a resources[] array, likely a subresource
-            // of a top-level resource.  This is possible if resources were copied into other
-            // resources as part of template expansion.
-            // TODO
+            // Get the line number of the original resource before it was copied
```

Comment:
Any reason the original comment can't stay here too?  It helps explain why the algorithm continues.